core/app/models/spree/stock_location.rb

```diff
@@ -108,10 +108,7 @@ def create_stock_items
 
       def ensure_one_default
         if self.default
-          StockLocation.where(default: true).where.not(id: self.id).each do |stock_location|
-            stock_location.default = false
-            stock_location.save!
-          end
+          StockLocation.where(default: true).where.not(id: self.id).update_all(default: false)
```

Comment:
@jasonfb Thanks for the feedback.
But, I am not sure that even with larger stores we are going to see more than few hundred's of stock locations, never mind only the defaults one on which update_all is running.(There would be only one default I think). We could have used update_column here in order to avoid callbacks, but then I am not big fan of 
```StockLocation.where(default: true).where.not(id: self.id).first.update_column()```